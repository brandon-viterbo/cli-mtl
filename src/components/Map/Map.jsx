import { useOutletContext } from "react-router-dom";

function PlaceCard({ place }) {
  return (
    <>
      <h3>{place.nom}</h3>
      <ul>
        <li>
          Adresse: {place.adresse_principale}, {place.ville}
        </li>
        <li>Arrondissement: {place.arrdondissement}</li>
        <li>Téléphone: {place.telephone}</li>
        <li>Accessibilité: {place.accessibilite}</li>
        <li>Lat: {place.lat}</li>
        <li>Long: {place.long}</li>
      </ul>
    </>
  );
}

function Map({ accessFilters, neighbourhoodFilters }) {
  const context = useOutletContext();
  let filterFunc;

  if (accessFilters.size === 0 && neighbourhoodFilters.size === 0) {
    filterFunc = (place) => true;
  } else if (accessFilters.size !== 0 && neighbourhoodFilters.size === 0) {
    filterFunc = (place) => accessFilters.has(place.properties.accessibilite);
  } else if (accessFilters.size === 0 && neighbourhoodFilters.size !== 0) {
    filterFunc = (place) =>
      neighbourhoodFilters.has(place.properties.arrdondissement);
  } else if (accessFilters.size !== 0 && neighbourhoodFilters.size !== 0) {
    filterFunc = (place) =>
      accessFilters.has(place.properties.accessibilite) &&
      neighbourhoodFilters.has(place.properties.arrdondissement);
  }

  const placeList = context.places.filter(filterFunc).map((place) => (
    <li key={place.id}>
      <PlaceCard place={place.properties} />
    </li>
  ));

  console.log(accessFilters.size === 0);
  console.log(neighbourhoodFilters.size === 0);

  return <ul>{placeList}</ul>;
}

export default Map;
