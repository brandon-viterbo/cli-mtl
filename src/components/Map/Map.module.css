.map {
  width: 100%;
  height: 550px;
  position: relative;
}

.places_list {
  left: 0;
  background-color: white;
  display: grid;
  grid-template-columns: minmax(320px, 500px);
  overflow: scroll;
  height: 550px;
  position: absolute;
  z-index: 1;
}

.places_list__close_button {
  border: none;
  border-radius: 10% 10% 10% 10%;
  background-color: red;
  color: white;
  text-align: center;
  position: absolute;
  top: 1rem;
  right: 1rem;
  z-index: 2;
}

.map > button {
  display: none;
  position: absolute;
  z-index: 1;
  bottom: 0px;
  right: 0px;
  margin-right: 1rem;
  margin-bottom: 1rem;
}

.places_list__card {
  border: 1px solid hsl(0deg, 0%, 70%);
  border-left: 0px;
  padding: 1rem;
}

.places_list span {
  font-weight: bold;
}

ul {
  list-style-type: none;
  padding: 0px;
}
