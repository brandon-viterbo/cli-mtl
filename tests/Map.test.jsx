import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import MockOutlet from "./MockOutlet";
import Map from "../src/components/Map/Map";

describe("Map component", () => {
  it("shows all places on initial visit", () => {
    const mockContext = {
      places: [
        {
          id: 0,
          properties: {
            nom: "A",
            adresse_principale: 0,
            ville: "Montréal",
            arrdondissement: "a_0",
            telephone: 0,
            accessibilite: "a",
            lat: 0,
            long: 0,
          },
        },
        {
          id: 1,
          properties: {
            nom: "B",
            adresse_principale: 1,
            ville: "Montréal",
            arrdondissement: "b_1",
            telephone: 1,
            accessibilite: "b",
            lat: 0,
            long: 0,
          },
        },
      ],
    };
    const { container } = render(
      <MockOutlet context={mockContext}>
        <Map accessFilters={new Set()} neighbourhoodFilters={new Set()} />
      </MockOutlet>,
    );
    expect(container).toMatchSnapshot();
  });

  it("filters on selected accessibility features", () => {
    const mockContext = {
      places: [
        {
          id: 0,
          properties: {
            nom: "A",
            adresse_principale: 0,
            ville: "Montréal",
            arrdondissement: "a_0",
            telephone: 0,
            accessibilite: "acc1",
            lat: 0,
            long: 0,
          },
        },
        {
          id: 1,
          properties: {
            nom: "B",
            adresse_principale: 1,
            ville: "Montréal",
            arrdondissement: "b_1",
            telephone: 1,
            accessibilite: "acc2",
            lat: 0,
            long: 0,
          },
        },
        {
          id: 2,
          properties: {
            nom: "C",
            adresse_principale: 2,
            ville: "Montréal",
            arrdondissement: "b_1",
            telephone: 0,
            accessibilite: "acc1",
            lat: 0,
            long: 0,
          },
        },
      ],
    };

    render(
      <MockOutlet context={mockContext}>
        <Map accessFilters={new Set(["acc1"])} neighbourhoodFilters={new Set()} />
      </MockOutlet>
    )

    expect(screen.queryAllByText("Accessibilité: acc1")).toHaveLength(2);
    expect(screen.queryByText("Accessibilité: acc2")).toBeNull();
  })

  it("filters on selected neighbourhoods", () => {
    const mockContext = {
      places: [
        {
          id: 0,
          properties: {
            nom: "A",
            adresse_principale: 0,
            ville: "Montréal",
            arrdondissement: "a_0",
            telephone: 0,
            accessibilite: "acc1",
            lat: 0,
            long: 0,
          },
        },
        {
          id: 1,
          properties: {
            nom: "B",
            adresse_principale: 1,
            ville: "Montréal",
            arrdondissement: "b_1",
            telephone: 1,
            accessibilite: "acc2",
            lat: 0,
            long: 0,
          },
        },
        {
          id: 2,
          properties: {
            nom: "C",
            adresse_principale: 2,
            ville: "Montréal",
            arrdondissement: "b_1",
            telephone: 0,
            accessibilite: "acc1",
            lat: 0,
            long: 0,
          },
        },
      ],
    };

    render(
      <MockOutlet context={mockContext}>
        <Map accessFilters={new Set()} neighbourhoodFilters={new Set(["a_0"])} />
      </MockOutlet>
    )

    expect(screen.queryAllByText("Arrondissement: a_0")).toHaveLength(1);
    expect(screen.queryByText("Arrondissement: b_1")).toBeNull();
  })

  it("filters on selected accessibility features and neighbourhoods", () => {
    const mockContext = {
      places: [
        {
          id: 0,
          properties: {
            nom: "A",
            adresse_principale: 0,
            ville: "Montréal",
            arrdondissement: "a_0",
            telephone: 0,
            accessibilite: "acc1",
            lat: 0,
            long: 0,
          },
        },
        {
          id: 1,
          properties: {
            nom: "B",
            adresse_principale: 1,
            ville: "Montréal",
            arrdondissement: "b_1",
            telephone: 1,
            accessibilite: "acc2",
            lat: 0,
            long: 0,
          },
        },
        {
          id: 2,
          properties: {
            nom: "C",
            adresse_principale: 2,
            ville: "Montréal",
            arrdondissement: "b_1",
            telephone: 0,
            accessibilite: "acc1",
            lat: 0,
            long: 0,
          },
        },
      ],
    };

    render(
      <MockOutlet context={mockContext}>
        <Map accessFilters={new Set(["acc2"])} neighbourhoodFilters={new Set(["b_1"])} />
      </MockOutlet>
    )

    expect(screen.queryAllByText("Arrondissement: b_1")).toHaveLength(1);
    expect(screen.queryByText("Accessibilité: acc1")).toBeNull();
  })
});
