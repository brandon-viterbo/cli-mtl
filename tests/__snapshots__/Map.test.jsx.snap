// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Map component > shows all places on initial visit 1`] = `
<div>
  <ul>
    <li>
      <h3>
        A
      </h3>
      <ul>
        <li>
          Adresse: 
          0
          , 
          Montréal
        </li>
        <li>
          Arrondissement: 
          a_0
        </li>
        <li>
          Téléphone: 
          0
        </li>
        <li>
          Accessibilité: 
          a
        </li>
        <li>
          Lat: 
          0
        </li>
        <li>
          Long: 
          0
        </li>
      </ul>
    </li>
    <li>
      <h3>
        B
      </h3>
      <ul>
        <li>
          Adresse: 
          1
          , 
          Montréal
        </li>
        <li>
          Arrondissement: 
          b_1
        </li>
        <li>
          Téléphone: 
          1
        </li>
        <li>
          Accessibilité: 
          b
        </li>
        <li>
          Lat: 
          0
        </li>
        <li>
          Long: 
          0
        </li>
      </ul>
    </li>
  </ul>
</div>
`;
